scylla-cql/src/errors.rs

```diff
@@ -40,6 +40,10 @@ pub enum QueryError {
 
     #[error("Unable to allocate stream id")]
     UnableToAllocStreamId,
+
+    /// Client timeout occurred before any response arrived
+    #[error("Client timeout: {0}")]
+    ClientTimeout(String),
```

Comment:
In at least one other driver (Python) this thing is called request_timeout, so let's call it that. It is a client-side timeout indeed, but request_timeout is more descriptive